perception/autoware_lidar_frnet/lib/lidar_frnet.cpp

```diff
@@ -0,0 +1,280 @@
+    RCLCPP_ERROR(
+      logger_,
+      "Number of points (%d) in input pointcloud does not match the profile. Consider to update "
+      "num_points in config file [min: %ld, opt: %ld, max: %ld].",
+      input_num_points, network_params_.num_points_profile.min,
+      network_params_.num_points_profile.opt, network_params_.num_points_profile.max);
+    return false;
+  }
+
+  auto cloud_in_d = cuda_utils::make_unique<InputPointType[]>(input_num_points);
+  CHECK_CUDA_ERROR(cudaMemcpyAsync(
+    cloud_in_d.get(), cloud_in.data.data(),
+    sizeof(std::uint8_t) * input_num_points * cloud_in.point_step, cudaMemcpyHostToDevice));
+
+  if (!preprocess(input_num_points, cloud_in_d)) {
+    RCLCPP_ERROR(logger_, "Preprocess failed.");
+    return false;
+  }
+  CHECK_CUDA_ERROR(cudaStreamSynchronize(stream_));
+  proc_timing.emplace(
+    "debug/processing_time/preprocess_ms", stop_watch_ptr_->toc("processing/inner", true));
+
+  if (!inference()) {
+    RCLCPP_ERROR(logger_, "Inference failed.");
+    return false;
+  }
+  CHECK_CUDA_ERROR(cudaStreamSynchronize(stream_));
+  proc_timing.emplace(
+    "debug/processing_time/inference_ms", stop_watch_ptr_->toc("processing/inner", true));
+
+  if (!postprocess(input_num_points, cloud_in_d, cloud_seg_out, cloud_viz_out, cloud_filtered)) {
+    RCLCPP_ERROR(logger_, "Postprocess failed.");
+    return false;
+  }
+  CHECK_CUDA_ERROR(cudaStreamSynchronize(stream_));
+  proc_timing.emplace(
+    "debug/processing_time/postprocess_ms", stop_watch_ptr_->toc("processing/inner", true));
+
+  return true;
+}
+
+bool LidarFRNet::preprocess(
+  const uint32_t input_num_points, CudaUniquePtr<InputPointType[]> & cloud_in_d)
+{
+  auto coors_keys_d = cuda_utils::make_unique<int64_t[]>(network_params_.num_points_profile.max);
+  auto num_points_d = cuda_utils::make_unique<uint32_t[]>(1);
+  auto proj_idxs_d = cuda_utils::make_unique<uint32_t[]>(
+    preprocessing_params_.interpolation.w * preprocessing_params_.interpolation.h);
+  auto proj_2d_d = cuda_utils::make_unique<float[]>(
+    preprocessing_params_.interpolation.w * preprocessing_params_.interpolation.h * 4);
+
+  cuda_utils::clear_async(points_d_.get(), network_params_.num_points_profile.max * 4, stream_);
+  cuda_utils::clear_async(coors_d_.get(), network_params_.num_points_profile.max * 3, stream_);
+  cuda_utils::clear_async(
+    voxel_coors_d_.get(), network_params_.num_unique_coors_profile.max * 3, stream_);
+  cuda_utils::clear_async(inverse_map_d_.get(), network_params_.num_points_profile.max, stream_);
+  cuda_utils::clear_async(coors_keys_d.get(), network_params_.num_points_profile.max, stream_);
+  cuda_utils::clear_async(num_points_d.get(), 1, stream_);
+  uint32_t num_unique_coors{0};
+
+  CHECK_CUDA_ERROR(cudaMemsetAsync(
+    proj_idxs_d.get(), 0,
+    preprocessing_params_.interpolation.w * preprocessing_params_.interpolation.h *
+      sizeof(uint32_t),
+    stream_));
+  CHECK_CUDA_ERROR(cudaMemsetAsync(
+    proj_2d_d.get(), -1.0,
```

Comment:
Just in case, is this intended behavior (signature-wise, this will not fill it with -1.0)

__host__​__device__​[cudaError_t](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1gf599e5b8b829ce7db0f5216928f6ecb6) cudaMemsetAsync ( void* devPtr, int  value, size_t count, [cudaStream_t](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1ge15d9c8b7a240312b533d6122558085a) stream = 0 )